src/notifications/notificationpreviewpresenter.cpp

```diff
@@ -203,9 +224,10 @@ void NotificationPreviewPresenter::setCurrentNotification(LipstickNotification *
 {
     if (currentNotification != notification) {
         if (currentNotification) {
-            const bool displayWasOn(currentNotification->hints().value(NotificationManager::HINT_DISPLAY_ON).toBool());
-            if (displayWasOn) {
-                // Release our screen wake
+            const bool notificationWasCritical = currentNotification->urgency() >= 2 ||
+                                                 currentNotification->hints().value(NotificationManager::HINT_DISPLAY_ON).toBool();
+            if (notificationWasCritical) {
```

Comment:
Could keep the var name as 'displayOn' since it is true even if the notification is not critical